apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadows.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+  SectionTitle,
+  SectionTitleButton,
+  SectionTitleLabel,
+} from "@webstudio-is/design-system";
+import { PlusIcon } from "@webstudio-is/icons";
+import { CollapsibleSectionBase } from "~/builder/shared/collapsible-section";
+import { useState } from "react";
+import { getDots } from "../../shared/collapsible-section";
+import { PropertyName } from "../../shared/property-name";
+import { getStyleSource } from "../../shared/style-info";
+import type { RenderCategoryProps } from "../../style-sections";
+import { BoxShadowLayersList } from "./box-shadow-list";
+import { addBoxShadow, property } from "./utils";
+import { parseBoxShadow } from "@webstudio-is/css-data";
+
+const label = "Box Shadow";
+
+export const BoxShadows = (props: RenderCategoryProps) => {
+  const { currentStyle, deleteProperty } = props;
+  const [isOpen, setIsOpen] = useState(true);
+  const layersStyleSource = getStyleSource(currentStyle[property]);
+  const value = currentStyle[property]?.value;
+
+  return (
+    <CollapsibleSectionBase
+      fullWidth
+      label={label}
+      isOpen={isOpen}
+      onOpenChange={setIsOpen}
+      trigger={
+        <SectionTitle
+          dots={getDots(currentStyle, [property])}
+          suffix={
+            <SectionTitleButton
+              prefix={<PlusIcon />}
+              onClick={() => {
+                const layers = parseBoxShadow(
+                  "0px 2px 5px 0px rgba(0, 0, 0, 0.2)"
+                );
+                // Will never be invalid.
+                if (layers.type === "invalid") {
+                  return;
+                }
+
+                if (value?.type === "layers") {
+                  // Adding layers we had before
+                  layers.value = [...layers.value, ...value.value];
```

Comment:
We have utils.ts with deleteLayer etc, obviously add layer should go there, with all additional checks etc like its done in backgrounds